### ISSUE TITLE
A map transform for finding last week’s (or whatever) value

### ISSUE THREAD

**Issue by mbostock** [observablehq/plot#1799]
With difference charts #159, we often want to compare the value today (or now) to the value some time period ago. Perhaps it would be simple enough, as with the window transform, to shift a column *n* slots forward or background.

**Comment by Fil** [comment]
I think this is `Plot.window({ anchor: "end", k: n + 1, reduce: "first" })`.
See https://observablehq.com/d/1e37e3c09876e460#withwindow

**Comment by mbostock** [comment]
That makes sense. Maybe we can make a more convenient/memorable form of that. (And probably can also optimize it?)

**Comment by Fil** [comment]
I'm not convinced we need an additional name for this particular transform (we could always add shiftX as an alias, but adding words has a cost too: documentation, more stuff to memorize, less generalization…). So imho let's try instead to make the existing feature more discoverable — maybe with a nice example notebook.

Also I've looked at how to optimize the first reducer further, but came empty-handed (unless we want to detect that there is only a single and complete series, and use Array.copy()?).

This issue's title makes me think though that sometimes we want to look up a certain channel (date) to derive another channel (value from the same series with the closest date to a week ago, with bisection for example), but it’s a different question? And readily addressable by using a function map on the identity channel (which I added to the notebook, in case we want to use it as a base for a tutorial). It feels a bit complicated, though.
